click/core.py

```diff
@@ -1289,6 +1290,8 @@ def __init__(self, param_decls=None, type=None, required=False,
         self.is_eager = is_eager
         self.metavar = metavar
         self.envvar = envvar
+        if autocompletion is not None:
```

Comment:
Nitpick but I would just allow this to be set to None here and then check `if cmd_param.autocompletion`  in _bashcomplete.py instead of hasattr. Or was there a reason that you did it this way that I'm missing?